client/src/main/java/io/pravega/client/stream/impl/ReaderGroupState.java

```diff
@@ -827,6 +827,11 @@ private void write00(SegmentCompleted object, RevisionDataOutput out) throws IOE
         @Override
         void update(ReaderGroupState state) {
             state.checkpointState.readerCheckpointed(checkpointId, readerId, positions);
+
+            // Each reader updates the offsets of its assigned segments with the current positions for this checkpoint.
+            for (Entry<Segment, Long> entry : positions.entrySet()) {
+                state.assignedSegments.get(readerId).computeIfPresent(entry.getKey(), (k, v) -> positions.get(entry.getKey()));
```

Comment:
state.assignedSegments.get(readerId) is exactly the same for every iteration of the loop. Move it out above.

Reply:
Done.